From 4c6af921e06c38a8f9ba9a3f0e945afef8559c91 Mon Sep 17 00:00:00 2001
From: Martin Pitt <martin.pitt@ubuntu.com>
Date: Wed, 3 Sep 2014 06:51:50 +0200
Subject: [PATCH] Fix Start TransientUnit signature for systemd >= 209

When building against systemd >= 209, add the new "aux" parameter to
StartTransientUnit().

https://launchpad.net/bugs/1351815
---
 configure.ac        | 3 +++
 src/systemd-iface.h | 5 +++++
 2 files changed, 8 insertions(+)

diff --git a/configure.ac b/configure.ac
index 98819da..e50cd12 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1,8 +1,11 @@
 AC_INIT([systemd-shim], [7])
 AM_INIT_AUTOMAKE([1.11 foreign -Wno-portability no-dist-gzip dist-xz])
+AC_CONFIG_HEADERS([config.h])
 AM_SILENT_RULES([yes])
 AC_PROG_CC
 PKG_CHECK_MODULES(gio, gio-2.0)
+PKG_CHECK_MODULES(systemd, systemd)
+AC_DEFINE_UNQUOTED([SYSTEMD_VERSION], [`$PKG_CONFIG --modversion systemd`], [systemd version])
 AC_CONFIG_FILES([Makefile
                  data/Makefile
                  src/Makefile])
diff --git a/src/systemd-iface.h b/src/systemd-iface.h
index 8c96695..d6c6944 100644
--- a/src/systemd-iface.h
+++ b/src/systemd-iface.h
@@ -1,6 +1,8 @@
 #ifndef _systemd_iface_h_
 #define _systemd_iface_h_
 
+#include "config.h"
+
 const gchar *systemd_iface =
   "<node>"
    "<interface name='org.freedesktop.systemd1.Manager'>"
@@ -35,6 +37,9 @@ const gchar *systemd_iface =
      "<arg name='name' type='s' direction='in'/>"
      "<arg name='mode' type='s' direction='in'/>"
      "<arg name='properties' type='a(sv)' direction='in'/>"
+#if SYSTEMD_VERSION >= 209
+     "<arg name='aux' type='a(sa(sv))' direction='in'/>"
+#endif
      "<arg name='job' type='o' direction='out'/>"
     "</method>"
     "<method name='Subscribe'/>"
-- 
2.1.0

